internal/testrunner/runners/system/tester.go

```diff
@@ -1987,10 +1995,78 @@ func (r *tester) checkTransforms(ctx context.Context, config *testConfig, pkgMan
 		if err != nil {
 			return fmt.Errorf("creating fields validator for data stream failed (path: %s): %w", transformRootPath, err)
 		}
-		if errs := validateFields(transformDocs, fieldsValidator); len(errs) > 0 {
-			return testrunner.ErrTestCaseFailed{
-				Reason:  fmt.Sprintf("errors found in documents of preview for transform %s for data stream %s", transformId, dataStream),
-				Details: errs.Error(),
+		if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == fieldsMethod {
+			if errs := validateFields(transformDocs, fieldsValidator); len(errs) > 0 {
+				return testrunner.ErrTestCaseFailed{
+					Reason:  fmt.Sprintf("errors found in documents of preview for transform %s for data stream %s", transformId, dataStream),
+					Details: errs.Error(),
+				}
+			}
+		}
+
+		if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == mappingsMethod {
+			destIndexTransform := transform.Definition.Dest.Index
+			// Index Template format is: "<type>-<package>.<transform>-template"
+			// For instance: "logs-ti_anomali.latest_intelligence-template"
+			indexTemplateTransform := fmt.Sprintf("%s-%s.%s-template",
+				ds.Inputs[0].Streams[0].DataStream.Type,
+				pkgManifest.Name,
+				transform.Name,
+			)
+
+			// TODO: use the docs obtained here for the fields validator too
```

Comment:
Do we need to check with the fields validator too if we are checking with the mappings? Or only if `allMethods` is true?

Reply:
Maybe this TODO is outdated (and could be removed), since there are some packages that could have just deleted documents in the destination index.

So, not sure if this would be possible to apply when the validation is based on fields :( 

WDYT ?

Reply:
As you prefer. My thought was this one, that we can just remove the TODO.
I guess fields validation is covered with the preview, and for mappings validation is enough with the changes in this PR.

Reply:
Agreed, I'll remove the TODO :+1: